test/onnx/test_op_consistancy.py

```diff
@@ -0,0 +1,365 @@
+    )
+
+
+def skip(
+    op_name: str,
+    variant_name: str = "",
+    *,
+    device_type: Optional[str] = None,
+    dtypes: Optional[Collection[torch.dtype]] = None,
+    reason="unspecified",
+):
+    """Skips a test case in OpInfo."""
+    return DecorateMeta(
+        op_name=op_name,
+        variant_name=variant_name,
+        decorator=unittest.skip(reason),
+        device_type=device_type,
+        dtypes=dtypes,
+        reason=reason,
+    )
+
+
+@dataclasses.dataclass
+class XfailOpset:
+    """Expects a OpInfo test to fail on specific ONNX opsets."""
+
+    op_name: str
+    opsets: Collection[Union[int, Callable[[int], bool]]]
+    dtypes: Optional[Collection[torch.dtype]] = None
+    exception: Optional[Exception] = None
+    reason: str = "unspecified"
+
+    def _contains_opset(self, opset: int) -> bool:
+        return any(
+            opset == opset_spec if isinstance(opset_spec, int) else opset_spec(opset)
+            for opset_spec in self.opsets
+        )
+
+    def _contains_dtype(self, dtype: torch.dtype) -> bool:
+        return self.dtypes is None or dtype in self.dtypes
+
+    def should_fail(self, opset: int, dtype: torch.dtype) -> bool:
+        """Returns whether the test should fail for the given opset and dtype."""
+        return self._contains_opset(opset) and self._contains_dtype(dtype)
+
+
+def skip_ops(
+    all_opinfos: Sequence[opinfo_core.OpInfo],
+    test_case_name: str,
+    base_test_name: str,
+    to_skip: Iterable[DecorateMeta],
+):
+    """Decorates OpInfo tests with decorators based on the to_skip list."""
+    ops_mapping = {(info.name, info.variant_test_name): info for info in all_opinfos}
+    for decorate_meta in to_skip:
+        opinfo = ops_mapping.get((decorate_meta.op_name, decorate_meta.variant_name))
+        assert (
+            opinfo is not None
+        ), f"Couldn't find OpInfo for {decorate_meta}. Did you need to specify variant_name?"
+        decorators = list(opinfo.decorators)
+        new_decorator = opinfo_core.DecorateInfo(
+            decorate_meta.decorator,
+            test_case_name,
+            base_test_name,
+            device_type=decorate_meta.device_type,
+            dtypes=decorate_meta.dtypes,
+        )
+        decorators.append(new_decorator)
+        opinfo.decorators = tuple(decorators)
+
+    # This decorator doesn't modify fn in any way
+    def wrapped(fn):
+        return fn
+
+    return wrapped
+
+
+def opsets_before(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is before the specified."""
+
+    def compare(other_opset: int):
+        return other_opset < opset
+
+    return compare
+
+
+def opsets_after(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is after the specified."""
+
+    def compare(other_opset: int):
+        return other_opset > opset
+
+    return compare
+
+
+# Modify this section ###
```

Comment:
nit: might want to make this more obvious, e.g. larger comment blocks, symbols or something like that

Reply:
Made it bigger